guides/source/threading_and_code_execution.md

```diff
@@ -106,60 +183,113 @@ ensure
+
+If the Executor is not already active on the current thread, the Reloader will
+invoke it for you, so you only need to call one. This also guarantees that
+everything the Reloader does, including all its callback executions, occurs
+wrapped inside the Executor.
 
 ```ruby
 Rails.application.reloader.wrap do
   # call application code here
 end
 ```
 
-The Reloader is only suitable where a long-running framework-level process
-repeatedly calls into application code, such as for a web server or job queue.
-Rails automatically wraps web requests and Active Job workers, so you'll rarely
-need to invoke the Reloader for yourself. Always consider whether the Executor
-is a better fit for your use case.
-
-### Callbacks
+#### Callbacks
 
 Before entering the wrapped block, the Reloader will check whether the running
-application needs to be reloaded -- for example, because a model's source file has
-been modified. If it determines a reload is required, it will wait until it's
-safe, and then do so, before continuing. When the application is configured to
-always reload regardless of whether any changes are detected, the reload is
-instead performed at the end of the block.
+application needs to be reloaded (because a model's source file has been
+modified, for example). If it determines a reload is required, it will wait
+until it's safe, and then do so, before continuing. When the application is
+configured to always reload regardless of whether any changes are detected, the
+reload is instead performed at the end of the block.
 
 The Reloader also provides `to_run` and `to_complete` callbacks; they are
 invoked at the same points as those of the Executor, but only when the current
 execution has initiated an application reload. When no reload is deemed
 necessary, the Reloader will invoke the wrapped block with no other callbacks.
 
-### Class Unload
+#### Class Unload
 
-The most significant part of the reloading process is the Class Unload, where
+The most significant part of the reloading process is the 'class unload', where
 all autoloaded classes are removed, ready to be loaded again. This will occur
-immediately before either the Run or Complete callback, depending on the
-`reload_classes_only_on_change` setting.
+immediately before either the `to_run` or `to_complete` callback, depending on
+the
+[`reload_classes_only_on_change`](configuring.html#config-reload-classes-only-on-change)
+setting.
 
 Often, additional reloading actions need to be performed either just before or
-just after the Class Unload, so the Reloader also provides `before_class_unload`
-and `after_class_unload` callbacks.
+just after the Class Unload, so the Reloader also provides
+[`before_class_unload`](https://api.rubyonrails.org/classes/ActiveSupport/Reloader.html#method-c-before_class_unload)
+and
+[`after_class_unload`](https://api.rubyonrails.org/classes/ActiveSupport/Reloader.html#method-c-after_class_unload)
+callbacks.
 
```

Comment:
It would be nice to show an code example here as well:
````suggestion

```ruby
Rails.application.reloader.before_class_unload do
  # call class unloading code here
end
```

````